src/every.ts

```diff
@@ -0,0 +1,63 @@
+import type { PredIndexed } from "./_types";
+import { purry } from "./purry";
+
+type TypePredIndexed<T, S extends T> = (
+  input: T,
+  index: number,
+  array: ReadonlyArray<T>,
+) => input is S;
```

Comment:
Inline this, in v2 all functions have the function operand stated explicitly, these make it harder to read the function signature to understand what it accepts.